File: systems-modelling/random-1/random/random.go
package random

var next int64 = 1

func Lcg() int {
	next = (next*1103515245 + 12345) % 2147483648
	return int(next)
}

func SetSeed(seed int64) {
	next = seed
}

func GenRandomSlice(len int) []float64 {
	res := make([]float64, len)

	for i := 0; i < len; i++ {
		res[i] = float64(Lcg())
	}

	return res
}

File: systems-modelling/random-1--/js/random.js
const RAND_MAX = 32767;
let next = Date.now();

export function lcg() {
   next = (1103515245 * next + 12345) % 2**31;
   return next;
}

export function setSeed(seed) {
   next = seed;
}

export function lcg2() {
   next = (1103515245 * next + 12345) % 2**31;
   return (next / 2**16) % (RAND_MAX + 1);
}